### DISCUSSION TITLE
Windows10 上屏时会崩溃

### DISCUSSION THREAD

**Discussion by Yiklek** [amorphobia/openfly#45] [一般讨论/General]
weasel版本：0.14.3
schema中注释掉`lua_processor@openfly_shortcut_processor`可正常上屏

崩溃时的日志
![image](https://user-images.githubusercontent.com/21033316/136665348-47719cf5-613e-4a94-a08e-7b302d14d665.png)

**Comment by lightyears1998** [comment]
@Yiklek  

我几个月前也遇到了这个问题。后来发现如 @amorphobia  在 README 的[“已知限制”小节](https://github.com/amorphobia/openfly#%E5%B7%B2%E7%9F%A5%E9%99%90%E5%88%B6)中提到的，由于 weasel v0.14.3 的 librime 版本还是太低，需要升级 librime 才能解决这个问题。可以参考 weasel repo 中的这个 [PR](https://github.com/rime/weasel/pull/624)。

我顺手 build 了一份，如果需要可以自取：~[weasel-0.14.3.0-installer.zip](https://github.com/amorphobia/openfly/files/7457655/weasel-0.14.3.0-installer.zip)~ [weasel-0.14.3.0-installer.zip](https://github.com/amorphobia/openfly/files/7487000/weasel-0.14.3.0-installer.zip)

**Comment by AlseinX** [comment]
这里是Windows 11，用的 @lightyears1998 的编译版本，全是这样的问题：
```
E1105 17:19:46.214826 14660 engine.cc:312] error creating processor: 'lua_processor'
E1105 17:19:46.219827 14660 engine.cc:346] error creating translator: 'lua_translator'
```
不会崩溃，但是所有的Lua就像不存在一样

**Comment by lightyears1998** [comment]
> 
> 
> 这里是Windows 11，用的 @lightyears1998 的编译版本，全是这样的问题：
> 
> ```
> E1105 17:19:46.214826 14660 engine.cc:312] error creating processor: 'lua_processor'
> E1105 17:19:46.219827 14660 engine.cc:346] error creating translator: 'lua_translator'
> ```
> 
> 不会崩溃，但是所有的Lua就像不存在一样

蚌埠住了呀。我我我忘记编译 `librime-lua` 了。这个版本应该就正常了。

[weasel-0.14.3.0-installer.zip](https://github.com/amorphobia/openfly/files/7487000/weasel-0.14.3.0-installer.zip)

**Comment by AlseinX** [comment]
@lightyears1998  OK了
话说我习惯官方版小鹤可以设置的那种让首选直接嵌入在输入的地方，然后候选区隐藏首选，在唯一候选时直接隐藏候选框，这个有办法做到吗

**Comment by lightyears1998** [comment]
> @lightyears1998 OK了 话说我习惯官方版小鹤可以设置的那种让首选直接嵌入在输入的地方，然后候选区隐藏首选，在唯一候选时直接隐藏候选框，这个有办法做到吗

不太清楚。

**Comment by amorphobia** [comment]
感谢 @lightyears1998 帮忙解答，这个词库一开始基本属于自己用，加上工作开始忙了，在我这里稳定之后就没怎么做更多测试了。

对于 lua 脚本我也是一知半解，可能这次是因为 `librime-lua` 版本问题，以后也会遇到脚本本身的 bug. 如果了解 lua 的，可以直接给我发 PR，谢谢！

@AlseinX 关于隐藏候选框，我记得我之前在网上找过，好像需要设置一个透明的主题，不过我不太需要，就没仔细了解。

**Reply by AlseinX** [reply]
关于那个透明主题，有相关的链接方便贴一下吗，我没找到噢

**Reply by amorphobia** [reply]
https://www.bilibili.com/video/BV1Gf4y167Lv

你看看这个呢

**Reply by AlseinX** [reply]
这个是mac的，有windows的吗

**Reply by amorphobia** [reply]
主题的设置应该都是类似的，仿照这个写一个应该可以的

**Reply by AlseinX** [reply]
噢噢我看看

**Reply by AlseinX** [reply]
好像不行，小狼毫不支持透明颜色
